MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm

```diff
@@ -332,7 +332,7 @@ static inline CIE1931XY VkXYColorEXTToCIE1931XY(VkXYColorEXT xy) {
 			// Nothing - the default is not to do color matching.
 			break;
 	}
-	_mtlLayerOrigDrawSize = _mtlLayer.updatedDrawableSizeMVK;
+	_mtlLayer.drawableSize = mvkCGSizeFromVkExtent2D(pCreateInfo->imageExtent);
```

Comment:
Interesting. Does this force an internal adjustment to the `contentsScale` value of the `CAMetalLayer`, and the system compositor just handles the scaling accordingly?

Reply:
No and yes. According to the docs:

> By default, a layer creates textures sized to match its content—that is, this property’s value is the layer’s `bounds` size multiplied by its `contentsScale` factor.

This property just affects the dimensions of the `MTLTexture`s produced. The system then handles scaling to the layer's content area. That's my understanding.

Reply:
Okay. Makes sense.

Just one more thing. Have you tested this on a Retina display? Part of the use of `contentsScale` handling was to handle Retina. But again...we may not have been interpreting its use correctly in the past.

Reply:
My laptop has a Retina display.

I think you might be confused. `bounds` and `contentsScale` just affect the coordinate system of the layer itself. Unless `drawableSize` is unset explicitly, they have no bearing on the size of the drawables that the layer produces.

This default size is returned as the current surface size, so this should work OK for Retina.

Reply:
Yeah. The original behaviour might have conflated the scaling of the layer coordinate system with that of the drawables.

Thanks for clearing this up.